### PR TITLE
OCPBUG#14320: Added a section related to oc-mirror supporting multi-arch images for…

### PR DESCRIPTION
**Pull request by subhtk**
Version(s): 4.14

Issue:
[OCPBUG14320](https://issues.redhat.com/browse/OCPBUGS-14320)

Link to docs preview: [Doc Preview](https://60959--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-14-release-notes.html#ocp-4-14-openshift-cli)

QE review:
- [x] QE has approved this change.

Additional information:
[CFE-783](https://github.com/openshift/oc-mirror/pull/611)

### CODE REVIEW

File: release_notes/ocp-4-14-release-notes.adoc [hunk 1]
```diff
@@ -57,6 +57,18 @@ This release adds improvements related to the following components and concepts.
...
 === Web console
+
+[id="ocp-4-14-openshift-cli"]
+=== OpenShift CLI (oc)
+
+.oc-mirror supporting multi-arch OCI local images for catalogs
+
+With {product-title} {product-version}, oc-mirror supports multi-arch OCI local images for catalogs.
+
+OCI layouts consist of an index.json file that identifies the images held within them on disk. This index.json file can reference any number of single or multi architecture images. However, oc mirror only references a single image at a time in a given OCI layout. The image stored in the OCI layout can be either a single architecture image (i.e. an image manifest) or a multi-architecture image (i.e. a manifest list).
```
Comment (zhouying7780) [inline release_notes/ocp-4-14-release-notes.adoc:68]: what's the meaning of "oc mirror only references a single image at a time in a given OCI layout" ?  Is it help to explain oc-mirror support the multi-arch images ?

File: release_notes/ocp-4-14-release-notes.adoc [hunk 2]
```diff
@@ -57,6 +57,18 @@ This release adds improvements related to the following components and concepts.
...
+
+The ImageSetConfiguration stores the OCI images, and after processing the catalog, the catalog content adds new layers representing the content of all images in the layout. The ImageBuilder is modified to handle image updates for both single-arch and multiarch images.
```
Comment (zhouying7780) [inline release_notes/ocp-4-14-release-notes.adoc:70]: better to change to "single-arch and multi-arch images"

File: release_notes/ocp-4-14-release-notes.adoc (hunk 1 above)
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:68]: Yeah, from this part here in  https://github.com/openshift/oc-mirror/pull/611 - "This index.json file can reference any number of single or multi architecture images, however oc mirror only references a single image at a time in a given OCI layout."

File: release_notes/ocp-4-14-release-notes.adoc (hunk 2 above)
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:70]: Ack.

Comment (zhouying7780) [reply release_notes/ocp-4-14-release-notes.adoc:70]: @subhtk we still see 'multiarch' here , better to use `multi-architecture`

File: release_notes/ocp-4-14-release-notes.adoc (hunk 1 above)
Comment (zhouying7780) [reply release_notes/ocp-4-14-release-notes.adoc:68]: @subhtk Got you , thanks

**Comment by tmalove** [comment]
@subhtk Will you add a preview link for your updates? Thanks.

**Comment by subhtk** [comment]
> @subhtk Will you add a preview link for your updates? Thanks.

The preview did not generate automatically. I am trying other means to get the the preview.

File: release_notes/ocp-4-14-release-notes.adoc [hunk 3]
```diff
@@ -57,6 +57,18 @@ This release adds improvements related to the following components and concepts.
 
 [id="ocp-4-14-web-console"]
 === Web console
+
+[id="ocp-4-14-openshift-cli"]
+=== OpenShift CLI (oc)
+
+.oc-mirror supporting multi-arch OCI local images for catalogs
```
Comment (tmalove) [inline release_notes/ocp-4-14-release-notes.adoc:64]: I suggest reworking the title to start with something like "Support multi-arch OCI..."

File: release_notes/ocp-4-14-release-notes.adoc [hunk 4]
```diff
@@ -57,6 +57,18 @@ This release adds improvements related to the following components and concepts.
...
+
+With {product-title} {product-version}, oc-mirror supports multi-arch OCI local images for catalogs.
+
+OCI layouts consist of an index.json file that identifies the images held within them on disk. This index.json file can reference any number of single or multi-arch images. However, oc-mirror only references a single image at a time in a given OCI layout. The image stored in the OCI layout can be either a single-arch image (i.e. an image manifest) or a multi-arch image (i.e. a manifest list).
+
+The ImageSetConfiguration stores the OCI images, and after processing the catalog, the catalog content adds new layers representing the content of all images in the layout. The ImageBuilder is modified to handle image updates for both single-arch and multi-arch images.
```
Comment (tmalove) [inline release_notes/ocp-4-14-release-notes.adoc:70]:
```suggestion
The `ImageSetConfiguration` stores the OCI images, and after processing the catalog, the catalog content adds new layers representing the content of all images in the layout. The ImageBuilder is modified to handle image updates for both single-arch and multi-arch images.
```

**Review by tmalove — APPROVED** [review]

File: release_notes/ocp-4-14-release-notes.adoc (hunk 3 above)
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:64]: @tmalove How about "Support multi-arch OCI local images for catalogs with oc-mirror"?

File: release_notes/ocp-4-14-release-notes.adoc [hunk 5]
```diff
@@ -57,6 +57,18 @@ This release adds improvements related to the following components and concepts.
...
+[id="ocp-4-14-openshift-cli"]
+=== OpenShift CLI (oc)
+
+.oc-mirror supporting multi-arch OCI local images for catalogs
+
+With {product-title} {product-version}, oc-mirror supports multi-arch OCI local images for catalogs.
+
+OCI layouts consist of an index.json file that identifies the images held within them on disk. This index.json file can reference any number of single or multi-arch images. However, oc mirror only references a single image at a time in a given OCI layout. The image stored in the OCI layout can be either a single-arch image (i.e. an image manifest) or a multi-arch image (i.e. a manifest list).
+
+The ImageSetConfiguration stores the OCI images, and after processing the catalog, the catalog content adds new layers representing the content of all images in the layout. The ImageBuilder is modified to handle image updates for both single-arch and multi-arch images.
```
[the remark this answers is not in the document]
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:70]: Ack.

File: release_notes/ocp-4-14-release-notes.adoc [hunk 6]
```diff
@@ -57,6 +57,18 @@ This release adds improvements related to the following components and concepts.
...
 === Web console
+
+[id="ocp-4-14-openshift-cli"]
+=== OpenShift CLI (oc)
+
+.oc-mirror supporting multi-arch OCI local images for catalogs
+
+With {product-title} {product-version}, oc-mirror supports multi-arch OCI local images for catalogs.
+
+OCI layouts consist of an index.json file that identifies the images held within them on disk. This index.json file can reference any number of single or multi-arch images. However, oc mirror only references a single image at a time in a given OCI layout. The image stored in the OCI layout can be either a single-arch image (i.e. an image manifest) or a multi-arch image (i.e. a manifest list).
```
[the remark this answers is not in the document]
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc]: Ack.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 4 above)
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:70]: Ack.

File: release_notes/ocp-4-14-release-notes.adoc [hunk 7]
```diff
@@ -63,6 +63,17 @@ User-defined tags for Microsoft Azure was previously introduced as Technology Pr
 [id="ocp-4-14-web-console"]
 === Web console
 
+[id="ocp-4-14-openshift-cli"]
+=== OpenShift CLI (oc)
+
+.Support multi-arch OCI local images for catalogs with oc-mirror
```
Comment (snarayan-redhat) [inline release_notes/ocp-4-14-release-notes.adoc:69]:
```suggestion
.Supporting multi-arch OCI local images for catalogs with oc-mirror
```

Here's the guideline:
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assemblymodule-titles-and-section-headings

File: release_notes/ocp-4-14-release-notes.adoc [hunk 8]
```diff
@@ -63,6 +63,17 @@ User-defined tags for Microsoft Azure was previously introduced as Technology Pr
...
+
+With {product-title} {product-version}, oc-mirror supports multi-arch OCI local images for catalogs.
+
+OCI layouts consist of an `index.json` file that identifies the images held within them on disk. This `index.json` file can reference any number of single or multi-arch images. However, oc-mirror only references a single image at a time in a given OCI layout. The image stored in the OCI layout can be a single-arch image (that is, an image manifest) or a multi-arch image (that is, a manifest list).
```
Comment (snarayan-redhat) [inline release_notes/ocp-4-14-release-notes.adoc:73]: As per ISG, avoid using parentheses in general text. I am even wondering whether `that is ...` is required.

```suggestion
OCI layouts consist of an `index.json` file that identifies the images held within them on disk. This `index.json` file can reference any number of single or multi-arch images. However, oc-mirror only references a single image at a time in a given OCI layout. The image stored in the OCI layout can be a single-arch image, that is, an image manifest or a multi-arch image, that is, a manifest list.
```

File: release_notes/ocp-4-14-release-notes.adoc [hunk 9]
```diff
@@ -63,6 +63,17 @@ User-defined tags for Microsoft Azure was previously introduced as Technology Pr
...
+
+The `ImageSetConfiguration` stores the OCI images, and after processing the catalog, the catalog content adds new layers representing the content of all images in the layout. The ImageBuilder is modified to handle image updates for both single-arch and multi-arch images.
```
Comment (snarayan-redhat) [inline release_notes/ocp-4-14-release-notes.adoc:75]: Is ImageBuilder an object? If yes, it needs to be in ``
```suggestion
The `ImageSetConfiguration` stores the OCI images. After processing the catalog, the catalog content adds new layers representing the content of all images in the layout. The ImageBuilder is modified to handle image updates for both single-arch and multi-arch images.
```

**Review by snarayan-redhat — COMMENTED** [review]
Few more changes that might be required before it can be merged.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 7 above)
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:69]: Ack.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 8 above)
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:73]: Ack.

File: release_notes/ocp-4-14-release-notes.adoc (hunk 9 above)
Comment (subhtk) [reply release_notes/ocp-4-14-release-notes.adoc:75]: Ack.

**Comment by subhtk** [comment]
@snarayan-redhat Done with changes, please go ahead with merging.

**Pull request merged**